code/game/machinery/atmos_exporter.dm

```diff
@@ -0,0 +1,278 @@
+		icon_state = "floorflush_a"
+		playsound(src, 'sound/machines/Ding.ogg', 100, 1)
+		spawn(20)
+			animate(C, transform = matrix() - matrix(), alpha = 0, color = rgb(0, 0, 0), time = 10)
+			for(var/t in 1 to 5)
+				if(!C || QDELETED(C))
+					return
+				C.pixel_y--
+				sleep(2)
+			icon_state = "floorflush_a2"
+			qdel(C)
+
+/obj/machinery/atmos_points/Initialize()
+	..()
+	e_gas += "Initializing"
+	s_gas += "Initializing"
+	s_gas += "Initializing"
+	p_gas += "Initializing"
+	p_gas += "Initializing"
+	p_gas += "Initializing"
+
+/obj/machinery/atmos_points/interact(mob/user)
+	user.set_machine(src)
+	for(var/obj/machinery/atmos_points_exporter/AE in machines)
+		can_turf = get_turf(AE)
+	var/dat = text("Time until market update: [round((market_cooldown - world.time)/10)] seconds<br>")
+	dat += "</div>"
+	dat += text("<A href='?src=\ref[src];action=send'>Export Canister</A> to <A href='?src=\ref[src];action=choose'> [export_name]</A><br>")
+	dat += text("<A href='?src=\ref[src];action=scan'>Scan Canister Contents</A><br><br>")
+	dat += "</div>"
+	dat += text("<u><b>Economy canister order for [economy_name]:</b></u><br>")
+	dat += text("Economy canister value: <b>[e_price] points</b><br>")
+	dat += text("Economy canister gas: <b>[e_gas[1]]</b><br>")
+	dat += text("Economy canister pressure range: <b>[e_pressure-200] to [e_pressure+200]kPa</b><br>")
+	dat += text("Economy canister temperature: <b>[e_temp-50] to [e_temp+50] kelvin</b><br>")
+	dat += "<br>"
+	dat += text("<u><b>Standard canister order for [standard_name]:</b></u><br>")
+	dat += text("Standard canister value: <b>[s_price] points</b><br>")
+	dat += text("Standard canister gas: <b>[s_gas[1]] at [(s_concentration1 - 5)]-[(s_concentration1 + 5)]%, [s_gas[2]] at [s_concentration2 - 5]-[s_concentration2 + 5]%</b><br>")
+	dat += text("Standard canister pressure range: <b>[s_pressure-100] to [s_pressure+100]kPa </b><br>")
+	dat += text("Standard canister temperature: <b>[s_temp-30] to [s_temp+30] kelvin</b><br>")
+	dat += "<br>"
+	dat += text("<u><b>Premium canister order for [premium_name]:</b></u><br>")
+	dat += text("Premium canister value: <b>[p_price] points</b><br>")
+	dat += text("Premium canister gas: <b>[p_gas[1]] at [(p_concentration1 - 5)]-[(p_concentration1 + 5)]%, [p_gas[2]] at [p_concentration2 - 5]-[p_concentration2 + 5]%, [p_gas[3]] at [p_concentration3 - 5]-[p_concentration3 + 5]%</b><br>")
+	dat += text("Premium canister pressure range: <b>[p_pressure-50] to [p_pressure+50] kPa </b><br>")
+	dat += text("Premium canister temperature: <b>[p_temp-15] to [p_temp+15] kelvin</b><br>")
+	var/datum/browser/popup = new(user, "vending", "Atmos Exporter", 400, 500)
+	popup.set_content(dat)
+	popup.open()
+	src.updateUsrDialog()
+
+/obj/machinery/atmos_points/Topic(href, href_list)
+	if(..())
+		return
+	switch(href_list["action"])
+		if ("choose")
+			reset()
+			export_target = input(usr, "Choose your export quality", "Quality:") as null|anything in quality

```

Comment:
after any spawn and sleep one must check if you are qdeleted or any of the things you're operating on are qdeleted

and this is a hidden sleep